pkgs/by-name/sa/satty/package.nix

```
@@ -0,0 +1,50 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, wrapGAppsHook4
+, cairo
+, gdk-pixbuf
+, glib
+, gtk4
+, libadwaita
+, pango
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "satty";
+  version = "unstable-2023-11-21";
+
+  src = fetchFromGitHub {
+    owner = "gabm";
+    repo = "Satty";
+    rev = "e1ad952e51c2f4a0f768d0d89418ae30eeda2cc3";
+    hash = "sha256-rpBOWZOwZcon8/09pAIqhBwU8uD6Nuh4gGjAzGFO0Y0=";
+  };
+
+  cargoHash = "sha256-UxFhs5I99kwo/+gY0k8Z20H8x1MJsG97SrswOqFJfAk=";
```

As discussed, please package the release version together with the cargo patch.
```suggestion
  version = "0.7.0";

  src = fetchFromGitHub {
    owner = "gabm";
    repo = "Satty";
    rev = "v${version}";
    hash = "sha256-x2ljheG7ZqaeiPersC/e8Er2jvk5TJs65Y3N1GjTiNU=";
  };

  cargoPatches = [
    (fetchpatch {
      name = "fix-Cargo.lock";
      url = "https://github.com/gabm/Satty/commit/39be6ddce264552df971e949a6a3175b102530b2.patch";
      hash = "sha256-GUHupZE1A7AmXvZ8WvRzBkQyH7qlMTetBjHuakfIZ7w=";
    })
  ];

  cargoHash = "sha256-0GsbWd/gpKZm7nNXkuJhB02YKUj3XCrSfpRA9KBXydU=";
```